platform/android/SCsub

```diff
@@ -95,25 +95,18 @@ if lib_arch_dir != "":
         else:
             gradle_process = ["./gradlew"]
 
-        if env["target"] != "editor" and env["dev_build"]:
-            subprocess.run(
-                gradle_process
-                + [
-                    "generateDevTemplate",
-                    "--quiet",
-                ],
-                cwd="platform/android/java",
-            )
-        else:
-            # Android editor with `dev_build=yes` is handled by the `generateGodotEditor` task.
-            subprocess.run(
-                gradle_process
-                + [
-                    "generateGodotEditor" if env["target"] == "editor" else "generateGodotTemplates",
-                    "--quiet",
-                ],
-                cwd="platform/android/java",
-            )
+        gradle_process += [
+            "generateGodotEditor" if env["target"] == "editor" else "generateGodotTemplates",
+            "--quiet",
+        ]
+
+        if env["dev_build"] or env["debug_symbols"]:
```

Comment:
`debug_symbols` is set implicitly when using `dev_build`, so checking only `debug_symbols` is enough.

In theory one could override the auto logic and do `scons dev_build=yes debug_symbols=no`, which would lead to a `DEV_ENABLED` build without debug symbols. That's a bit pointless, but in that case there would also be little point set `doNotStrip`.
```suggestion
        if env["debug_symbols"]:
```